### PR TITLE
Fix interfaces support in Merge API

### LINKED ISSUE #2388: Problem with Merge operation when using interfaces
Hello there!

This code on previous version worked fine (I've tested 3.0.0 preview.2):

```cs
public ITable<IReviewIndex> ReviewIndexes => GetTable<ReviewIndex>();
...

await db.ReviewIndexes
    .Merge()
    .Using(items)
    .On(x => new { x.Index, x.Date }, x => new { x.Index, x.Date })
    .UpdateWhenMatched()
    .InsertWhenNotMatched()
    .MergeAsync();
```
Now with version 3.0.0 and later it failed with wrong mapping error System.ArgumentException Property 'System.Decimal Value' is not defined for type 'Finam.Interfaces.Data.Entities.Comon.IReviewIndex' (Parameter 'property')

Mappings declared in ReviewIndex class which implements IReviewIndex interface. Other operations (select, insert, delete etc) work as expected.

### Environment details
linq2db version: 3.0.1
Database Server: MS SQL Server
Database Provider: MSSQL
Operating system: Windows 10
.NET Framework: .NET Core 3.0

### PR DESCRIPTION
**Pull request by MaceWindu**
Fix #2388

### CODE REVIEW

**Comment by MaceWindu** [comment]
Case from issue is fixed, but plan to add tests for more merge methods to make sure it works everywhere

File: Source/LinqToDB/Linq/Builder/MergeBuilder.cs [hunk 1]
```diff
@@ -87,9 +87,12 @@ public static SqlSearchCondition BuildSearchCondition(ExpressionBuilder builder,
 				{
 					var secondContextRefExpression =
 							new ContextRefExpression(condition.Parameters[1].Type, secondContext);
+					var targetParameter = Expression.Parameter(tableContext.ObjectType);
 
-					var newBody = condition.Body.Replace(condition.Parameters[1], secondContextRefExpression);
-					condition   = Expression.Lambda(newBody, condition.Parameters[0]);
+					var newBody = condition.Body
+						.Replace(condition.Parameters[1], secondContextRefExpression)
+						.Replace(condition.Parameters[0], targetParameter);
```
Comment (sdanyliv) [inline Source/LinqToDB/Linq/Builder/MergeBuilder.cs:94]:
```suggestion
					var newBody = condition.GetBody(targetParameter, secondContextRefExpression);
```

**Pull request merged**